models/order.js

```diff
@@ -12,7 +13,68 @@ const Entry = new Schema({
 const orderSchema = new Schema({
   id: String,
   status: { type: String, default: constants.OPEN },
-  items: [Entry]
+  entries: [Entry]
 });
 
+orderSchema.statics.getOrders = function() {
+  return getOrders.apply(this)
+  .then(getEntries)
+  .catch((e) => {
+    throw new Error(e);
+  });
+}
+
+function getOrders() {
+  const Order = this;
+
+  return new Promise((resolve, reject) => {
+    Order.find().exec((err, orders) => {
+      if(err) {
+        reject(err);
+      }
+
+      resolve(orders.map(toOrder));
+    });
+  });
+}
+
+function populateEntries(order) {
+  return new Promise((resolve, reject) => {
+    Item.populate(order.entries, [{ path: 'item_id', model: 'Item', select: 'name price -_id' }], (err, res) => {
```

Comment:
I can't think of how to do this. I'm struggling with the structure of the orders - so not all the populateEntries promise is on the same level...does that make any sense? And I don't know if Promise.all is able to resolve it, hence having to through the orders, and assigning the result of promise.all to each order's entries. sorry, that was a mouth full.

Orders: [
  {id, entries: [{item_id: PROMISE, comment, status}]},
  {id, entries: [{item_id: PROMISE, comment, status}]},
]

 If you are okay with it, I'm fine with this for now, since it's working.